piptools/resolver.py

```diff
@@ -289,6 +291,15 @@ def _iter_dependencies(self, ireq):
         dependency_strings = self.dependency_cache[ireq]
         log.debug('  {:25} requires {}'.format(format_requirement(ireq),
                                                ', '.join(sorted(dependency_strings, key=lambda s: s.lower())) or '-'))
+
+        if self.exclude:
+            dep_set = {str(self.get_best_match(InstallRequirement.from_line(e))) for e in dependency_strings}
```

Comment:
Calling `get_best_match` here means that the lookup on PyPi will still be done, and an error will occur if the packages isn't found. One of the main use case I see of `--exclude` would be to deal with the lack of proper environment marker handling we have right now, but it would't work in this case.

Is there a way to get the same logic without doing the lookup? Alternatively, catching and ignoring the `NoCandidateFound` error could be enough for now.